### DISCUSSION TITLE
/hyndsight/arima-trends/

### DISCUSSION THREAD

**Discussion by robjhyndman** [robjhyndman/robjhyndman.com#175] [Announcements]
Testing for trend in ARIMA models 
 https://robjhyndman.com/hyndsight/arima-trends/

**Comment by robjhyndman** [comment]
```header
Originally posted by ezra abrams on 2014-03-12 23:50:47
````
if 1 person doesn\'t get statistics, that is the person\
if most people most of the time don\'t get statistics, it is the
teachers

**Comment by robjhyndman** [comment]
```header
Originally posted by Tao Hong on 2014-03-13 01:55:41
````
I would probably make a different recommendation to the authors who run
into ignorance reviewers. Instead of trying to please the reviewer, I
think they should write down in their response letter how nonsense the
review is (with solid evidence of course), and request the journal to
ban this reviewer. If the EIC is a domain expert in the same field,
write the letter to him/her as well.

**Comment by robjhyndman** [comment]
```header
Originally posted by Guest on 2014-03-13 10:16:07
````
Thank you a lot for the article. Maybe you have an example with R and
can share it. I would be truly appreciated.

**Comment by robjhyndman** [comment]
```header
Originally posted by Brian on 2014-03-14 01:08:56
````
One point of confusion for a novice like me: If you have a model of
Arima(0,1,1) shouldn\'t the trend by definition be zero? Else, doesn\'t
that imply that you needed another round of differencing to achieve
stationary series to then use the Arima? I guess I am wondering if the
constant (mu above) doesnt simply represent the &quot;trend&quot; that remained
after a single differencing?

**Reply by robjhyndman** [reply]
```header
Originally posted on 2014-03-14 05:44:01
````
No. An ARIMA(0,1,1) model is not stationary, and if it has a constant in
the model, then the data will show a trend. The constant represents the
mean of the differenced data, or the slope of the trend in the
undifferenced data.

**Reply by robjhyndman** [reply]
```header
Originally posted by Brian on 2014-03-14 18:58:12
````
Thanks!

**Reply by robjhyndman** [reply]
```header
Originally posted by Jeffery K on 2014-07-11 01:10:05
````
Isn\'t an ARIMA(0,1,1) stationary? I thought that was the whole point of
the &quot;I&quot;? Maybe I am misunderstanding?

**Reply by robjhyndman** [reply]
```header
Originally posted by Jeffery K on 2014-07-11 13:33:43
````
Let me articulate my confusion better. If we have a series {y}\_t and
fit an ARIMA(0,1,1) to that data, assuming it fits well, would we not
say that the series has been differenced (once) and is now stationary? I
understand if you difference the series {y}\_t and this differenced
series is not centered around zero, then the mean of the differenced
series needs to be taken into acocunt and the constant will do that. But
the statement that ARIMA(0,1,1) fit to data, seemed to me to imply it
corrected the non-stationary data to now be stationary. Am I more
inexperienced with this than I even thought? :)

**Reply by robjhyndman** [reply]
```header
Originally posted on 2014-07-14 22:47:32
````
It is stationary after differencing. It is not stationary before
differencing. The ARIMA(0,1,1) model is applied to the undifferenced
data, so it is not a stationary model.

**Comment by robjhyndman** [comment]
```header
Originally posted by Marta on 2014-03-14 08:19:28
````
Dear Rob J Hyndman,\
Many thanks for the article and your job. Maybe you could explain step
by step please, how to detect an increasing trend with arima(0,1,1) ?
Does it work only on increasing trend, or on decreasing too? Thank you a
lot in advance!

**Comment by robjhyndman** [comment]
```header
Originally posted by Barry Quinn on 2014-03-30 11:03:17
````
Rob thanks a lot for the really useful post. In many financial
econometrics books they distinguish between a stochastic trend
(y=A+By(t-1)+u) and deterministic trend (y=A+B(t)+u). Although both are
trending over time, differencing is recommended to induce stationarity
in the former while applying differencing to the latter introduces an
ma(1) term (and is thought not to be required as the process is trend
stationary). Instead it is suggested to use the residuals from this
regression,y=A+B(t)+u, as a detrended series in further analysis. Excuse
my ignorance but given your explanation above is this distinction in
treatment irrelevant ?

**Reply by robjhyndman** [reply]
```header
Originally posted on 2014-03-30 21:53:24
````
That distinction is useful. See
[http://www.otexts.org/fpp/9/1](http://www.otexts.org/fpp/9/1)
for a discussion of the two models.

**Reply by robjhyndman** [reply]
```header
Originally posted by Barry Quinn on 2014-03-31 08:08:48
````
Thanks Rob, a much clearer exposition.

**Comment by robjhyndman** [comment]
```header
Originally posted by Eugene Tuan on 2017-05-11 20:49:26
````
I have a question. The equation indicates a drift term. What about the
constant term?

**Reply by robjhyndman** [reply]
```header
Originally posted on 2017-05-11 22:40:12
````
They are equivalent. See
[https://www.otexts.org/fpp2\...](https://www.otexts.org/fpp2/arima-modelling-in-r.html#eq:mu &quot;https://www.otexts.org/fpp2/arima-modelling-in-r.html#eq:mu&quot;)

**Reply by robjhyndman** [reply]
```header
Originally posted by Eugene Tuan on 2017-05-13 05:34:13
````
Rob, thanks for your reply. But I still have some confusion. I would
appreciate if you could advise again.

1\. If the constant term and drift term are equivalent, then why R Arima
has two parameters, include.constant and include.drift? Moreover, the
link in the following defines the term &quot;delta \* t&quot; as &quot;time trend&quot;
and constant C as constant term (or drift term). Is the definition about
&quot;constant&quot; and &quot;drift&quot; the same as R Arima?

[https://stats.stackexchange\...](https://stats.stackexchange.com/questions/104215/difference-between-series-with-drift-and-series-with-trend &quot;https://stats.stackexchange.com/questions/104215/difference-between-series-with-drift-and-series-with-trend&quot;)

2\. However, my guess is R Arima doesn\'t handle the linear time trend
term. Am I correct? If it\'s true, do you have idea why?

3\. If I write my own Java code for Arima, how do I calculate the
constant and/or coefficient for the time trend term? My difficulty is
after making the time series stationary through differencing, both time
trend and constant properties have been removed.

4\. Can you please to see enclosed image for my equation? In this
equation, I have both constant term and drift term. But that\'s
different from your equation. Is my equation still correct?

[https://uploads.disquscdn.c\...](https://uploads.disquscdn.com/images/07369943b166bf993212e64cb38ce4b280168c324fb959ff67c70c0672b86c14.png)

**Reply by robjhyndman** [reply]
```header
Originally posted on 2017-05-13 05:49:03
````
1\. include.constant is more general. It will include a constant in any
Arima model, include.drift is only relevant when d=1. include.drift is
retained for backwards compatibility.

2\. You can handle a linear trend by including a time covariate via
xreg.

3\. The time trend becomes a constant after differencing, it is not
removed.

4\. If the AR polynomial has no unit root, then you don\'t have a drift
term \-- you have a linear trend.

If the AR polynomial does have a unit root of order d, then you actually
have a forecast function in the form of a polynomial of order d+1.

In neither case do you have a simple drift term.

The second last equation is incorrect.

**Reply by robjhyndman** [reply]
```header
Originally posted by Eugene Tuan on 2017-05-13 16:18:25
````
Thank you very much. Would you minding answering question 1 to question
3? Thanks a lot again

**Reply by robjhyndman** [reply]
```header
Originally posted by Eugene Tuan on 2017-05-19 19:42:34
````
Thank you very much. It is helpful. I have two followup questions.

1\) If the original time series (refer to the bottom image) has c1 term
and c2t term, after differencing, the c1 term goes away. But how do we
calculate c1 in order to plug in c1 value to the original equation for
prediction? Is c1 not important?

2\) From user standpoint, under which circumstance the include.mean
should be used or should not be used?

[https://uploads.disquscdn.c\...](https://uploads.disquscdn.com/images/83decafe6da79f7cc743c3280c60cfb4dcd7111d9300ff8c290a11761c302e18.png)

**Reply by robjhyndman** [reply]
```header
Originally posted on 2017-05-19 22:44:49
````
c1 is not identifiable, so just set it to 0

Read
[https://robjhyndman.com/hyn\...](https://robjhyndman.com/hyndsight/arimaconstants/)

**Comment by robjhyndman** [comment]
```header
Originally posted by Eugene Tuan on 2017-07-13 00:14:05
````
Rob,

I have a couple of questions. Right now I am developing our own Varmax
function. Do you think the drift term for Varmax is important as well
like Arima?

Thanks

Eugene

**Comment by robjhyndman** [comment]
```header
Originally posted by Eugene Tuan on 2017-08-29 04:40:52
````
Rob, I have a question about the drift term again. I know how to
calculate if there is only auto-regressive part. But I don\'t know how
to calculate for moving average part. For auto-regressive part, (Y_t -
c) = phi \* (Y_t - c) + e. What about moving average part?

Assuming the model is (0, 1, 1). Should I say (after differencing): Y_t
= e + theta \* e_t-1 + c, where c is the drift?

What about (1, 1, 1)? Is this correct after differencing?

(Y_t - c) = phi \* (Y_t - c) + e + theta \* e_t-1?

Thanks

**Reply by robjhyndman** [reply]
```header
Originally posted on 2017-08-29 08:05:35
````
See
[http://otexts.org/fpp2/arim\...](http://otexts.org/fpp2/arima-modelling-in-r.html#understanding-constants-in-r &quot;http://otexts.org/fpp2/arima-modelling-in-r.html#understanding-constants-in-r&quot;)

**Comment by robjhyndman** [comment]
```header
Originally posted by Sneha on 2017-09-01 21:54:11
````
How to force coefficient sign for xreg variable in auto.arima() ? I have
two dummy variables, each showing the weeks with positive and negative
impact due to campaign launches, etc. in my model. I need to force the
coefficient signs to be positive and negative respectively. The model
selected by auto.arima() currently reverses the signage of these
variables\' coefficients.

**Reply by robjhyndman** [reply]
```header
Originally posted on 2017-09-02 01:41:54
````
You would need to write your own code for that.

**Comment by robjhyndman** [comment]
```header
Originally posted by joshua hemmingway on 2018-08-02 14:28:23
````
Rob, can you comment on the efficacy of trend tests in the \'trend\'
package in R?

**Reply by robjhyndman** [reply]
```header
Originally posted on 2018-08-02 23:14:02
````
No, I\'m not very familiar with the package.

**Comment by robjhyndman** [comment]
```header
Originally posted by edct40 on 2019-04-12 00:06:42
````
Great explanation! is there any possibility to obtain the statistical
significance of the trend of the arima model?

**Reply by robjhyndman** [reply]
```header
Originally posted on 2019-04-12 00:46:46
````
Read the second last paragraph.

**Comment by MuhaNel** [comment]
Hi,
i am a first time user of the ARIMA model and i would like to find out how determine/detect seasonal trends using the ARIMA model

**Reply by robjhyndman** [reply]
An ARIMA model has an implicit trend if the number of differences (d+D) is 2 and there is no constant, or if the number of differences (d+D) is 1 and there is a constant. It includes seasonality if any part of the seasonal components is included. i.e., D+P+Q>0.

If your goal is to detect seasonality or trends, an ETS model would be better, as seasonality and trend are explicitly modelled.

**Reply by MuhaNel** [reply]
Thank you very much will try the model.
